### PR TITLE
Fixes run button appearing when it shouldn't

### LINKED ISSUE #36621: rustdoc: Renders Run button even if playpen is not enabled
rustdoc renders the **Run** button even if playpen is not enabled now. Clicking the button has no useful effect.

Was this caused by #36334 ?

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #36621.

r? @steveklabnik

### CODE REVIEW

**Comment by bluss** [comment]
Thanks!

File: src/librustdoc/html/static/playpen.js [hunk 1]
```diff
@@ -15,6 +15,11 @@ document.addEventListener('DOMContentLoaded', function() {
     'use strict';
 
     if (!window.playgroundUrl) {
+        var run_buttons = document.querySelectorAll(".test-arrow");
```
Comment (frewsxcv) [inline src/librustdoc/html/static/playpen.js]: Very very minor, but it looks like naming conventions in the rest of this file use camel case, not snake case.

Comment (GuillaumeGomez) [inline src/librustdoc/html/static/playpen.js]: It's javascript, who cares. :trollface:

Comment (frewsxcv) [inline src/librustdoc/html/static/playpen.js]: I think there is some value in caring about keeping consistent with naming conventions. It's not clear to me why we shouldn't care about JavaScript.

Comment (GuillaumeGomez) [inline src/librustdoc/html/static/playpen.js]: I used trollface to make sure people understand it was a troll. I failed :-/

I'll update later on.

**Comment by GuillaumeGomez** [comment]
Updated.

**Review by frewsxcv — APPROVED** [review]

**Comment by bluss** [comment]
I tried this locally and this does not fix the issue. playpen.js is not used at all for a crate without playgroundUrl configured.

See first and last lines in this range: https://github.com/rust-lang/rust/blob/289f3a4ca79916d6445b452fc19a18a1e42a879a/src/librustdoc/html/layout.rs#L143-L181

**Comment by GuillaumeGomez** [comment]
Strange. I'll retest once again. Thanks for your feedback @bluss!

**Comment by GuillaumeGomez** [comment]
@bluss: Thanks for pointing me to this! Can you test it again please?

**Comment by bluss** [comment]
@GuillaumeGomez  The fix looks like it will work to me. I would probably just put it in main.js, that simplifies things. Unfortunately I'd have to rebuild llvm to test the rustdoc build, and I don't have the time to do it.

**Comment by GuillaumeGomez** [comment]
I don't want it all the time, I just want it in case playpen.js isn't present.

**Comment by GuillaumeGomez** [comment]
Hum... I'll create another js file instead.

**Comment by GuillaumeGomez** [comment]
Ok, seems like it's all good now.

**Comment by steveklabnik** [comment]
@bluss what do you think of this PR now?

**Comment by bluss** [comment]
It looks like it will work fine. But since both extra.js, and playpen.js have the same snippet to remove the run buttons, that part, it might as well be in main.js instead?

**Comment by GuillaumeGomez** [comment]
@bluss: No, the point is to avoid putting too much watchers. One by button is already enough. If you put this part in main.js, it means we'll have it duplicated since playpen.js (in case it'll be there) will still put the watcher on it.

**Comment by bluss** [comment]
Oh I see, that makes sense.

**Comment by bluss** [comment]
@bors r+

Let's go.

**Pull request merged**